Add plain Jenkins support (#22)

cosmin-malutan opened the pull request:
This is for issue #22.
I added setup and start scrips, git ignore files and config.xml.
hudson.model.UpdateCenter.xml so it will point to the stable version

cosmin-malutan committed:
Add jenkins for coversheet (#22)

cosmin-malutan commented:
@whimboo I missed to tag you here :) can you please have look?
this is just plain jenkins with git ignore

whimboo commented on .gitignore:
```diff
@@ -1,3 +1,8 @@
+*.war
+log/
 *.swp
 *.egg-info
```
Please use the same entries as we have in the mozmill-ci repository. We don't need anything else. Some are still missing.

whimboo commented on jenkins-master/.gitignore:
```diff
@@ -0,0 +1,49 @@
+plugins/ssh-slaves.jpi
+plugins/subversion.jpi
+plugins/translation.jpi
+plugins/windows-slaves.jpi
```
Why are you adding all of those plugins? We don't know yet which ones will be in use, so you can only add those which come by default with Jenkins.

cosmin-malutan replied:
At the first ./start.py it will install those plugins and they would appear for the commit, those come as default plugins.

whimboo replied:
Ok, then it is fine.

whimboo commented on jenkins-master/envInject.xml:
```diff
@@ -0,0 +1,4 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<org.jenkinsci.plugins.envinject.EnvInjectNodeProperty_-EnvInjectNodePropertyDescriptor plugin="envinject@1.88">
+  <envInjectGlobalPasswordEntries/>
+</org.jenkinsci.plugins.envinject.EnvInjectNodeProperty_-EnvInjectNodePropertyDescriptor>
```
When you add this file you would also have to add the appropriate plugin. We aren't doing this yet.

whimboo commented on setup.sh:
```diff
@@ -0,0 +1,45 @@
+fi
+
+echo "Installing required dependencies"
+pip install --upgrade --global-option="--pure" mercurial==${VERSION_MERCURIAL}
```
Why do you install mercurial?

whimboo commented on start.py:
```diff
@@ -0,0 +1,63 @@
+
+    # TODO: Start Jenkins as daemon
+    print "Starting Jenkins"
+    os.environ['JENKINS_HOME'] = 'jenkins-master'
```
This code is broken. Haven't you seen the latest fixes in mozmill-ci? Please update and watch the repository for changes.

whimboo commented:
Beside the comments I miss the test infrastructure, and the basic tests. We need those.

cosmin-malutan commented:
@whimboo  I added the test structure from mozmill-ci, here you have to be careful to install the latest phanthomjs and addit to the path, and don't rely on Ubuntu software package installer which will install the 1.6 version.

whimboo commented:
When you see problems with a tool we are using, please give details why it is important to care about a specific version. The above comment only raises an alert without giving any details to why this is important. So what's the problem with the 1.6 phantomjs package as can be found in Ubuntu?

whimboo commented on jenkins-master/config.xml:
```diff
@@ -0,0 +1,48 @@
+  <buildsDir>${ITEM_ROOTDIR}/builds</buildsDir>
+  <jdks/>
+  <viewsTabBar class="hudson.views.DefaultViewsTabBar"/>
+  <myViewsTabBar class="hudson.views.DefaultMyViewsTabBar"/>
```
You silently added all those lines above in your last commit without giving any feedback on it. Did you miss it the last time or why those appear now?

cosmin-malutan replied:
This is the default configuration, the file gets created if it doesn't exist when we run the test(save_config.py), since this it's not under gitignore as it will hold the future configuration, I want to add it as part of this commit so it won't confuse other collaborators that will ran the tests and then will make a commit.

whimboo replied:
I know that this is the default configuration. I do not talk about the file itself, but the latest changes you did in the last commit. Please read my comments carefully.

cosmin-malutan replied:
Those should be in the default config config.xml, I removed them manually when I moved the code from mozmill-ci but it looks like the starting of jenkins adds those nodes in config file.

whimboo replied:
That looks fine then.

whimboo commented on tests/configuration/__init__.py:
```diff
@@ -0,0 +1 @@
+__author__ = 'cosmin'
```
Why that file? Please remove that.

whimboo commented:
Can you please also add the Readme file with the basic setup steps and commands to use? A lot of it is similar to the Mozmill-CI one.

cosmin-malutan commented:
Regarding phanthomjs I'v got the following exception when running tests:

> selenium.common.exceptions.WebDriverException: Message: 'Can not connect to GhostDriver' 

I followed the indication from the link bellow and I installed the latest version of it and it worked for me, I didn't troubled to find out what's  wrong with the 1.6 version.

> http://stackoverflow.com/questions/20309170/selenium-and-phantomjs-error-cannot-connect-to-ghostdriver#answer-22321284

cosmin-malutan commented:
I added the README from mozmill-ci for setting up details, I updated it to be relevant on coversheet, I removed the extra file I added in the previous commit.

whimboo commented:
Something I don't see yet is the removal of all the old coversheet files, which we do not need anytime longer. We should not carry them around with us.

cosmin-malutan commented:
I removed the old coversheet files.

whimboo commented:
This PR has merge conflicts.

cosmin-malutan commented:
@whimboo I made a rebase, I had conflicts in removing the the old coversheet files which have changed.

whimboo commented:
There is still the config subfolder existent. Why have you kept this? It's nothing we need on the master.

cosmin-malutan commented:
I thought we would still use that config untill we would implement a queuing api for accounts creations, but you're right it shouldn't live on master anyway, I removed it.

whimboo commented:
The tests are failing due to uncommited changes, and not added files. Aren't you running the tests and checking for not tracked files via `git status`?

cosmin-malutan commented:
For me the test passes, it prints a diff at the end with:

```
--- a/jenkins-master/queue.xml
+++ /dev/null
@@ -1,2 +0,0 @@
-<?xml version='1.0' encoding='UTF-8'?>
-<list/>
\ No newline at end of file
```

But queue.xml is in gitignore, I shouldn't add it in the commit, I'm not sure what changes causes  this failure. 
Probably we get that because we ran "Saving job configurations... " while we have 0 jobs, if so this should pass when we will create jobs.

whimboo commented:
The `queue.xml` file is part of your patch. So if that file should NOT be part of the repository, remove it! Otherwise Travis will mark this as a test failure. So please always check what you commit with `git status` before.

cosmin-malutan commented:
I removed them and I didn't saw that diff at the end.

whimboo commented:
Cosmin, can you please read my review comment VERY carefully? As already said in https://github.com/mozilla/coversheet/pull/23#issuecomment-46428652 you are missing files!

cosmin-malutan commented:
I was confused which files I was missing here since on mozmill-ci we also have uncommitted files after running the tests, and I wasn't aware that travis checks that because locally I just ran the sh script.
I added all the created files on jenkins-master and the other related to virtualenv I added to gitignore.

whimboo commented:
No, we do have any files which are needed by Jenkins core as part of the repository. Not sure to which files you actually referring to. If you see inconsistencies you should file an issue on mozmill-ci.

cosmin-malutan commented:
The virtual environment which gets downloaded while running tests it's untracked.
I created issue https://github.com/mozilla/mozmill-ci/issues/452 for this.
If no other issues I would squash the commits.

whimboo commented:
This looks fine now and can be rebased to master.

cosmin-malutan commented:
@whimboo done

whimboo commented:
This PR has been merged with the newly created https://github.com/mozilla/coversheet/tree/jenkins branch.
